crates/evm/src/evm/handler.rs

```diff
@@ -25,16 +25,39 @@ use crate::{BASE_FEE_VAULT, L1_FEE_VAULT};
 
 const DB_ACCOUNT_SIZE: usize = 256;
 
-// Normally db account key is: 24 bytes of prefix + 1 byte for size of remaining data + 20 bytes of address = 45 bytes
-// But we already add address size to diff size, so we don't need to add it here
+/// Normally db account key is: 24 bytes of prefix + 1 byte for size of remaining data + 20 bytes of address = 45 bytes
+/// But we already add address size to diff size, so we don't need to add it here
 const DB_ACCOUNT_KEY_SIZE: usize = 25;
 
+/// Storage key is 77 bytes because of sov sdk prefix
+const STORAGE_KEY_SIZE: usize = 77;
+
+/// Storage value is 33 bytes because of 1 extra byte of size descriptor at the beginning of the value of StateMap
+const STORAGE_VALUE_SIZE: usize = 33;
+
 /// We write data to da besides account and code data like block hashes, pending transactions and some other state variables that are in modules: evm, soft_confirmation_rule_enforcer and sov_accounts
 /// The L1 fee overhead is to compensate for the data written to da that is not accounted for in the diff size
 /// It is calculated by measuring the state diff we write to da in a single batch every 10 minutes which is about 300 soft confirmations
 /// The full calculation can be found here: https://github.com/chainwayxyz/citrea/blob/erce/l1-fee-overhead-calculations/l1_fee_overhead.md
 pub const L1_FEE_OVERHEAD: usize = 4;
 
+/// These ratios are calculated because we want to charge the user for the amount of data they write to the state
+/// But in order to calculate that we need to know the size of the state diff which is not possible to calculate in a single transaction
+/// So we calculate these ratios to estimate the total diff size that would end up in a batch
+/*
+Let's consider a batch of 1 block with the following transactions:
+
+    Block 1:
+        Transaction 1: Account A transfers balance to Account C
+        Transaction 2: Account B transfers balance to Account C
+
+    In this account A and B pays for the balance state diff of C, but at the end of the batch the diffs are merged and there is one state diff for C
+    So A and B should share that cost thus we have the ratios below
+*/
+const NONCE_DISCOUNTED_PERCENTAGE: usize = 55;
```

Comment:
add comment stating that nonce and balance are stored together so we use single constant